homeassistant/components/meteoclimatic/__init__.py

```diff
@@ -0,0 +1,103 @@
+"""Support for Meteoclimatic weather data."""
+import asyncio
+import datetime
+import logging
+
+from meteoclimatic import MeteoclimaticClient
+from meteoclimatic.exceptions import MeteoclimaticError
+import voluptuous as vol
+
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.typing import ConfigType, HomeAssistantType
+from homeassistant.util import Throttle
+
+from .const import CONF_STATION_CODE, DOMAIN, PLATFORMS
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = datetime.timedelta(minutes=5)
+
+
+STATION_CODE_SCHEMA = vol.Schema({vol.Required(CONF_STATION_CODE): cv.string})
+
+CONFIG_SCHEMA = vol.Schema(
+    {DOMAIN: vol.Schema(vol.All(cv.ensure_list, [STATION_CODE_SCHEMA]))},
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+async def async_setup(hass: HomeAssistantType, config: ConfigType) -> bool:
+    """Set up Meteoclimatic from legacy config file."""
+
+    conf = config.get(DOMAIN)
+    if conf is None:
+        return True
+
+    for station_code_conf in conf:
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": SOURCE_IMPORT}, data=station_code_conf.copy()
+            )
+        )
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistantType, entry: ConfigEntry) -> bool:
+    """Set up an Meteoclimatic account from a config entry."""
+    hass.data.setdefault(DOMAIN, {})
+
+    # Weather
+    station_code = entry.data[CONF_STATION_CODE]
+    client = MeteoclimaticClient()
+    hass.data[DOMAIN][station_code] = MeteoclimaticUpdater(client, station_code)
+    await hass.async_add_executor_job(hass.data[DOMAIN][station_code].update)
```

Comment:
What happens if this fails?

Reply:
I'm now capturing the `MeteoclimaticError` in the `MeteoclimaticUpdater._update()` method.